src/expr/src/vector_op/concat_op.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::array::{BytesGuard, BytesWriter};
+
+use crate::Result;
+
+#[inline(always)]
+pub fn concat_op(left: &str, right: &str, dst: BytesWriter) -> Result<BytesGuard> {
+    dst.write_from_char_iter(left.chars().chain(right.chars()))
+        .map_err(Into::into)
```

Comment:
Calling `write_ref` twice is supposedly more efficient than going thru char iter. 😂

Reply:
Seems that the bytes writer disallows us to write twice by taking the ownership. 😢

Reply:
begin

Reply:
Cool.